walk-in-interview/src/main/webapp/update-job/script.js

```diff
@@ -368,21 +374,62 @@ submitButton.addEventListener('click', (_) => {
         headers: {'Content-Type': 'application/json'},
         body: JSON.stringify(jobDetails),
     })
-        .then((response) => response.text())
-        .then((data) => {
-            console.log('data', data);
-            /** reset the error (there might have been an error msg from earlier) */
-            setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ '', /* includesDefault= */false);
-            window.location.href= HOMEPAGE_PATH;
+        .then((response) => {
+          if (response.status == BAD_REQUEST_STATUS_CODE) {
+            setErrorMessage(/* errorMessageElementId= */'error-message',
+              /* msg= */ UPDATE_JOB_STRINGS['storing-error-message'], /* includesDefault= */false);
+            throw new Error(UPDATE_JOB_STRINGS['storing-error-message']);
+          }
+
+          /** reset the error (there might have been an error msg from earlier) */
+          setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ '', /* includesDefault= */false);
+          window.location.href= HOMEPAGE_PATH;
         })
         .catch((error) => {
-            setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ RESPONSE_ERROR,
-                /* includesDefault= */false);
+          // Not the server response error already caught and thrown
+          if (error.message != UPDATE_JOB_STRINGS['storing-error-message']) {
+            setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ UPDATE_JOB_STRINGS['error-message'],
+              /* includesDefault= */false);
             console.log('error', error);
+          }
         });
 });
 
 const cancelButton = document.getElementById('cancel');
 cancelButton.addEventListener('click', (_) => {
     window.location.href= HOMEPAGE_PATH;
 });
+
+/**
+ * Returns a dummy jobId only for selenium ui test.
+ */
+function getJobIdForSeleniumTest() {
+  return "xxxxxxx";
+}
+
+/**
+ * Returns a specific job object only for selenium ui test.
+ */
+function getJobForSeleniumTest() {
```

Comment:
why do you need this? Shouldn't you be testing existing functions

Reply:
Since the GET request to get job from servlet is not implemented yet, so i simply return a pre-defined job object to test